Examples/trame/app.py

```diff
@@ -0,0 +1,305 @@
+    print(f"rate_of_change: {rate_of_change}")
+
+
+def reset_rate_of_change():
+    state.rate_of_change = DEFAULT_RATE_OF_CHANGE
+
+
+# -----------------------------------------------------------------------------
+# GUI Cards
+# -----------------------------------------------------------------------------
+
+compact_style = {
+    "hide_details": True,
+    "dense": True,
+}
+
+
+def ui_card(title, ui_name):
+    card_style = {}
+    title_style = {
+        "classes": "grey lighten-1 py-1 grey--text text--darken-3",
+        "style": "user-select: none; cursor: pointer",
+        **compact_style,
+    }
+    content_style = {"classes": "py-2"}
+    with vuetify.VCard(**card_style):
+        vuetify.VCardTitle(title, **title_style)
+        content = vuetify.VCardText(**content_style)
+
+    return content
+
+def input_card():
+    with ui_card("Input", "input"):
+        vuetify.VFileInput(
+            multiple=True,
+            show_size=True,
+            small_chips=True,
+            truncate_length=25,
+            v_model=("files", None),
+            dense=True,
+            hide_details=True,
+            style="max-width: 300px;",
+            accept=".nrrd,.vti",
+            __properties=["accept"],
+        )
+    
+    # TODO Add a component allowing to re-order "state.input_file_names"
+    # Possible approaches:
+    # - v-simple-table + SortableJS
+    #   See https://codepen.io/mykysyk/pen/qBdBRMB
+    # - v-chip + draggable
+    #   See https://github.com/vuetifyjs/vuetify/issues/11614 and https://jsfiddle.net/cjrqzkf0/
+
+
+def sizing_field_row(**kwargs):
+    with vuetify.VRow():
+        vuetify.VSlider(
+            **kwargs,
+            thumb_label=True,
+            classes="my-1",
+            **compact_style,
+        )
+        with vuetify.VBtn(icon=True, click=reset_sampling_rate, **compact_style):
+            vuetify.VIcon("mdi-restore")
+
+
+def sizing_field_card():
+    with ui_card("Sizing Field", "sizing-field"):
+        sizing_field_row(
+            v_model=("sampling_rate", DEFAULT_SAMPLING_RATE),
+            min=0.01,
+            max=99.9,
+            step=1.0,
+            label="Sampling Rate"
+        )
+        sizing_field_row(
+            v_model=("feature_scaling", DEFAULT_FEATURE_SCALING),
+            min=0.0001,
+            max=9999.0,
+            step=0.0001,
+            label="Feature Scaling"
+        )
+        sizing_field_row(
+            v_model=("rate_of_change", DEFAULT_RATE_OF_CHANGE),
+            min=0.0001,
+            max=10.0,
+            step=0.0001,
+            label="Rate of Change (Lipschitz)"
+        )
+
+
+def cleaving_tool_card():
+    with ui_card("Cleaving Tool", "cleaving-tool"):
+        with vuetify.VRow(justify="center", **compact_style):
+            vuetify.VBtn(
+                "Cleave Mesh",
+                disabled=("!input_available || cleaver_running",),
+                loading=("cleaver_running", True),
```

Comment:
The state of the button is not updated, I was not able to "force" the state to be updated on the client.

@jourdain  Would appreciate any suggestion.

Reply:
Probably because your server is busy working. You need to make it async and run the C++ exec in a task so you can flush the state before your work is done.

Reply:
This is what I was thinking, I will then apply the approach used in https://github.com/Kitware/trame-mnist/blob/master/trame_mnist/app/engine/main.py by using `ProcessPoolExecutor` ...

I naively thought updating the state object would be sufficient ...

Would it make sense to have the management of state run in a dedicated process by default?

Reply:
It is a tricky decision, but I have been thinking of it and I'm not sure yet on how to properly do it. BTW, you do not need `ProcessPoolExecutor`. Just a task could be enough.